src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,497 @@
+
+Amazon OpenSearch Ingestion, combined with S3 exports and DynamoDB streams, facilitates seamless data input from DynamoDB tables and automatic ingestion into OpenSearch. Additionally, the pipeline can back up data to S3 for potential future re-ingestion as needed.
+
+## Step 1: Setup the project
+Begin by setting up your project by following the instructions in the [Quickstart guide](/[platform]/start/quickstart/). For the purpose of this guide, we'll sync a Todo table from DynamoDB to OpenSearch.
+
+Firstly, add the Todo model to your schema:
+
+```ts title="amplify/data/resource.ts"
+
+import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
+
+const schema = a.schema({
+  // highlight-start
+  Todo: a
+    .model({
+      content: a.string(),
+      done: a.boolean(),
+      priority: a.enum(["low", "medium", "high"]),
+    })
+    .authorization((allow) => [allow.publicApiKey()])
+  // highlight-end  
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: "apiKey",
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+
+```
+
+Important considerations:
+
+Ensure Point in Time Recovery (PITR) is enabled, which is crucial for the pipeline integration.
+Enable DynamoDB streams to capture item changes that will be ingested into OpenSearch.
+
+```ts title="amplify/backend.ts"
+
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+// highlight-start
+const todoTable =
+  backend.data.resources.cfnResources.amplifyDynamoDbTables["Todo"];
+
+// Update table settings
+todoTable.pointInTimeRecoveryEnabled = true;
+
+todoTable.streamSpecification = {
+  streamViewType: dynamodb.StreamViewType.NEW_IMAGE,
+};
+
+// Get the DynamoDB table ARN
+const tableArn = backend.data.resources.tables["Todo"].tableArn;
+// Get the DynamoDB table name
+const tableName = backend.data.resources.tables["Todo"].tableName;
+// highlight-end 
+
+```
+
+## Step 2: Setting Up the OpenSearch Instance
+
+Create an OpenSearch instance with encryption.
+
+``` ts title="amplify/backend.ts"
+
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+import * as opensearch from "aws-cdk-lib/aws-opensearchservice";
+import * as osis from "aws-cdk-lib/aws-osis";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { RemovalPolicy } from "aws-cdk-lib";
+import {  Stack } from "aws-cdk-lib";
+
+
+// Get the data stack
+const openSearchStack = Stack.of(backend.data);
```

Comment:
Would it be confusing to call the data stack as `openSearchStack`?

Reply:
changed to datastack